target/scripts/helpers/variables.sh

```diff
@@ -0,0 +1,193 @@
+#! /bin/bash
+
+# shellcheck disable=SC2034
+declare -A VARS
+
+# shellcheck disable=SC2034
+declare -a FUNCS_SETUP FUNCS_FIX FUNCS_CHECK FUNCS_MISC DAEMONS_START
+
+function _environment_variables_early_setup
+{
+  # setup supervisord as early as possible
+
+  VARS[SUPERVISOR_LOGLEVEL]="${SUPERVISOR_LOGLEVEL:=warn}"
+
+  if [[ $(type -t _setup_supervisor) == 'function' ]]
+  then
+    _setup_supervisor
+  else
+    _log 'error' "Early variable setup missing function '_setup_supervisor', was probably not sourced correctly beforehand"
+  fi
+
+
+  # handle DNS names
+
+  if [[ $(type -t _obtain_hostname_and_domainname) == 'function' ]]
+  then
+    _obtain_hostname_and_domainname
+  else
+    _log 'error' "Early variable setup missing function '_obtain_hostname_and_domainname', was probably not sourced correctly beforehand"
+  fi
```

Comment:
I did this because of the entanglement of setting envs and calling these functions. I resolved this, and this code can therefore be omitted completely :) Thanks for pointing this out.